### PR TITLE
Added maven profile to run a set of integration tests for the new image certification

### PR DESCRIPTION
**Pull request by hzhao-github**
this set of tests added to the new maven profile
ItParameterizedDomain.java
ItServerStartPolicy.java
ItMiiUpdateDomainConfig.java
ItIntrospectVersion.java
ItMiiSample.java,
ItJrfDomainInPV.java,
ItStickySession.java
ItSessionMigration.java

based on the QA wiki:
http://aseng-wiki.us.oracle.com/asengwiki/display/WLSQA/WebLogic+Kubernetes+Operator+Certification

Jenkins job passed

1. parallel + fmw-image-cert
https://build.weblogick8s.org:8443/job/wko-tmp/92

2. parallel + wls-image-cert
https://build.weblogick8s.org:8443/job/wko-tmp/95/

3. parallel + integration-tests
https://build.weblogick8s.org:8443/job/wko-tmp/94/

### CODE REVIEW

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
Can you modify kindtest.sh to use PROFILE env variable if its set, if not use integration-tests as default profile to run. We can add a new parameter in Jenkins for profile.

File: kindtest.sh [hunk 1]
```diff
@@ -73,6 +75,7 @@ parallel_run="false"
 threads="2"
 wdt_download_url="https://github.com/oracle/weblogic-deploy-tooling/releases/latest"
 wit_download_url="https://github.com/oracle/weblogic-image-tool/releases/latest"
+verify_image_run="false"
```
Comment (vanajamukkara) [inline kindtest.sh:78]:
```suggestion
maven_profile_name="integration-tests"
```

File: kindtest.sh [hunk 2]
```diff
@@ -94,6 +97,8 @@ while getopts ":h:n:o:t:v:c:x:p:d:i:" opt; do
     ;;
     i) wit_download_url="${OPTARG}"
     ;;
+    m) verify_image_run="${OPTARG}"
```
Comment (vanajamukkara) [inline kindtest.sh:100]:
```suggestion
    m) maven_profile_name="${OPTARG}"
```

**Review by anpanigr — COMMENTED** [review]
We should add two profiles  (wls-image-cert  and fmw-image-cert ) to certify WebLogic Image and FMW images respectively.

File: kindtest.sh [hunk 3]
```diff
@@ -237,9 +242,19 @@ rm -rf "${RESULT_ROOT:?}/*"
 
 echo 'Run tests...'
 if [ "${test_filter}" = "ItOperatorUpgrade" ] || [ "${parallel_run}" = "false" ]; then
-  echo "Running mvn -Dit.test=${test_filter} -Dwdt.download.url=${wdt_download_url} -Dwit.download.url=${wit_download_url} -pl integration-tests -P integration-tests verify"
-  time mvn -Dit.test="${test_filter}" -Dwdt.download.url="${wdt_download_url}" -Dwit.download.url="${wit_download_url}" -pl integration-tests -P integration-tests verify 2>&1 | tee "${RESULT_ROOT}/kindtest.log"
+  if [ "${verify_image_run}" != "verify-images-tests" ]; then
```
Comment (vanajamukkara) [inline kindtest.sh:245]: you don't need this condition, use ${maven_profile_name} in the mvn command

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: kindtest.sh (hunk 3 above)
Comment (hzhao-github) [reply kindtest.sh]: true. Thanks

File: kindtest.sh (hunk 1 above)
Comment (hzhao-github) [reply kindtest.sh]: fixed

File: kindtest.sh (hunk 2 above)
Comment (hzhao-github) [reply kindtest.sh]: fixed

File: kindtest.sh (hunk 3 above)
Comment (hzhao-github) [reply kindtest.sh]: fixed

File: kindtest.sh [hunk 4]
```diff
@@ -235,11 +242,18 @@ export JAVA_HOME="${JAVA_HOME:-`type -p java|xargs readlink -f|xargs dirname|xar
 echo 'Clean up result root...'
 rm -rf "${RESULT_ROOT:?}/*"
 
-echo 'Run tests...'
+echo 'Determine test filter ...'
+if [ "${maven_profile_name}" = "wls-image-cert" ]; then
+  test_filter="${test_filter_wls}"
+elif [ "${maven_profile_name}" = "fmw-image-cert" ]; then
+  test_filter="${test_filter_fmw}"
+fi
```
Comment (anpanigr) [inline kindtest.sh:250]: if (maven_profile_name)  is not defined  (the current behaviour) what path it should follows ?  It should default to current behavior  where it does not take any maven profile.

File: kindtest.sh [hunk 5]
```diff
@@ -68,13 +70,16 @@ else
   outdir="${WORKSPACE}/logdir/${BUILD_TAG}"
 fi
 test_filter="**/It*"
+test_filter_wls="**/ItParameterizedDomain.java,**/ItServerStartPolicy.java,**/ItMiiUpdateDomainConfig.java,**/ItIntrospectVersion.java,**/ItMiiSample.java,**/ItJrfDomainInPV.java,**/ItStickySession.java,**/ItSessionMigration.java"
```
Comment (vanajamukkara) [inline kindtest.sh:73]: these are defined in the profile, you don't have to set here

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: kindtest.sh (hunk 4 above)
Comment (hzhao-github) [reply kindtest.sh:250]: default is test_filter="**/It*" in line 72

**Review by vanajamukkara — APPROVED** [review]
LG

**Comment by anpanigr** [comment]
Can u run the test once in https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new  to make sure it does not break the existing jenkin configuration

**Review by anpanigr — APPROVED** [review]
LGTM

**Pull request merged**